### DISCUSSION TITLE
Cart = null using Frontend API and a logged in user

### DISCUSSION THREAD

**Discussion by soyzamudio** [swellstores/community#303] [Help] [answered]
I'm using the FE Api and trying to get a Cart, I know there's an item in the cart for the current x-session, which I see store in the cookies, but whenever I do `swell.cart.get()`, I get null as the response.

Do i have to pass anything to the `.get()` call?

**Comment by logeshswell** [comment]
No, it is not required to pass any arguments as mentioned in the doc:
https://developers.swell.is/frontend-api/carts#get-a-cart

I assume that the `x-session` might not be passed correctly, which is one of the main reasons for finding an empty cart always.

Can you share the sample code and framework that you use?

**Reply by soyzamudio** [reply]
I'm using nextj.js 13, this is the code to get the cart:
```ts
import swell from "swell-js";

// Initialize the client first
swell.init(SWELL_URL, PUBLIC_KEY, {
  useCamelCase: true,
});

export function getCart() {
  return swell.cart.get();
}
```

and this is where I'm calling that function:

```tsx
import { getCart } from "@lib/cart";

async function getCartInformation() {
  const res = await getCart();
  console.log(res); // logging null
  return res;
}

const CartPage = async () => {
  const cart = await getCartInformation();
  console.log("cart", cart); // logging null

  return (
    <div className="container mx-auto">
      <h1 className="font-fancy text-4xl">Cart: {cart?.id}</h1>
    </div>
  );
};

export default CartPage;
```

**Reply by logeshswell** [reply, marked as answer]
@soyzamudio I tried the same code in my system but in an angular app and it works fine as expected.

Can you cross-check once whether `X-Session` header when you add an item to the cart is the same as the one in fetching the cart?

You can find this information in the network tab of the browser.

**Reply by logeshswell** [reply]
I assume that the cart sessions are not persisted properly in the subsequent requests.

**Comment by soyzamudio** [comment]
is there a reason this got marked as closed? :(

**Reply by logeshswell** [reply]
When we need some additional information relevant to the discussion, we mark it as Closed and add `need-more-info` label, until it is provided.

It can be reopened by either the author or us when the relevant information is provided.
